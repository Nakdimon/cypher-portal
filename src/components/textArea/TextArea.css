.cypher-input-text {
    height: 80px;
    width: 300px;
    border-radius: 5px;
    font-family: inherit;
    font: 400 13.3333px Arial;
    resize: none;      
  }