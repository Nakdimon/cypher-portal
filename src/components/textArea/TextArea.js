import React from 'react';
import './TextArea.css'

function TextArea({
    isTextAreaDisabled,
    handleTextAreaChange
}) {
    return (
        <textarea disabled={isTextAreaDisabled} className="cypher-input-text" onChange={handleTextAreaChange} placeholder="Input the text you want to cypher."></textarea>
    )

}

export default TextArea;