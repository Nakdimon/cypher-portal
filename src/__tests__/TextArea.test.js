import React from 'react';
import TextArea from '../components/textArea/TextArea';
import { shallow }  from '../enzyme'

const handleTextAreaChange = jest.fn()

describe("Text area component: ", ()=> {
  it("Matches the snapshot", () => {
    const component = shallow(
      <TextArea
        isTextAreaDisabled={false}
        handleTextAreaChange={handleTextAreaChange}
      />
    )
    expect(component).toMatchSnapshot();
  }),

  it("Renders", () => {
    const component = shallow(
      <TextArea
        isTextAreaDisabled={false}
        handleTextAreaChange={handleTextAreaChange}
      />
    )
    expect(component.exists()).toBe(true)
  }),

  it("Runs handler when changed", () => {
    const component = shallow (
        <TextArea
          isTextAreaDisabled={false}
          handleTextAreaChange={handleTextAreaChange}
        />
      )
    
    const event = { 
        preventDefault() {}, 
        target: {value: 'change'} 
    }

    component.find('textarea').simulate('change', event)
    
    expect(handleTextAreaChange).toBeCalledWith(event);
  })
})