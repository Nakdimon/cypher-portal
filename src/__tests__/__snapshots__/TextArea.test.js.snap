// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component:  Matches the snapshot 1`] = `
<textarea
  className="cypher-input-text"
  disabled={false}
  onChange={[Function]}
  placeholder="Input the text you want to cypher."
/>
`;

exports[`Text area component:  Matches the snapshot 1`] = `ShallowWrapper {}`;
